docs/create_your_own_brick.rst

```diff
@@ -0,0 +1,387 @@
+    ...         self.input_dim1 = input_dim1
+    ...         self.input_dim2 = input_dim2
+    ...         self.output_dim1 = output_dim1
+    ...         self.output_dim2 = output_dim2
+    ...
+    ...     def __allocate(self, input_dim, output_dim, number):
+    ...         W = shared_floatx_nans((input_dim, output_dim),
+    ...                                name='W'+number)
+    ...         add_role(W, WEIGHT)
+    ...         self.parameters.append(W)
+    ...         self.add_auxiliary_variable(W.norm(2), name='W'+number+'_norm')
+    ...
+    ...     def _allocate(self):
+    ...         self.__allocate(self.input_dim1, self.output_dim1, '1')
+    ...         self.__allocate(self.input_dim2, self.output_dim2, '2')
+    ...
+    ...     def _initialize(self):
+    ...         W1, W2 = self.parameters
+    ...         self.weights_init.initialize(W1, self.rng)
+    ...         self.weights_init.initialize(W2, self.rng)
+    ...
+    ...     @application(inputs=['input1_', 'input2_'], outputs=['output1',
+    ...         'output2'])
+    ...     def apply(self, input1_, input2_):
+    ...         """Apply the two linear transformations.
+    ...
+    ...         Parameters
+    ...         ----------
+    ...         input{1,2}_ : :class:`~tensor.TensorVariable`
+    ...             The two inputs on which to apply the transformations
+    ...
+    ...         Returns
+    ...         -------
+    ...         output{1,2} : :class:`~tensor.TensorVariable`
+    ...             The two inputs multiplied by their respective matrices
+    ...
+    ...         """
+    ...         W1, W2 = self.parameters
+    ...         output1 = tensor.dot(input1_, W1)
+    ...         output2 = tensor.dot(input2_, W2)
+    ...         return output1, output2
+    ...
+    ...     def get_dim(self, name):
+    ...         if name == 'input1_':
+    ...             return self.input_dim1
+    ...         if name == 'input2_':
+    ...             return self.input_dim2
+    ...         if name == 'output1':
+    ...             return self.output_dim1
+    ...         if name == 'output2':
+    ...             return self.output_dim2
+    ...         super(ParallelLinear, self).get_dim(name)
+
+You can test the brick as follows:
+
+   >>> input_dim1, input_dim2, output_dim1, output_dim2 = 10, 5, 2, 1
+   >>> batch_size1, batch_size2 = 1, 2
+   >>>
+   >>> x1_mat = 3 * numpy.ones((batch_size1, input_dim1),
+   ...                         dtype=theano.config.floatX)
+   >>> x2_mat = 4 * numpy.ones((batch_size2, input_dim2),
+   ...                         dtype=theano.config.floatX)
+   >>>
+   >>> x1 = theano.tensor.matrix('x1')
+   >>> x2 = theano.tensor.matrix('x2')
+   >>> parallel1 = ParallelLinear(input_dim1, input_dim2, output_dim1,
+   ...                            output_dim2, weights_init=Constant(2))
+   >>> parallel1.initialize()
+   >>> # The weights_init initialization scheme is pushed to the children
```

Comment:
This statement should be illustrated by actually checking `weight_init` of the children.